core/trino-main/src/main/java/io/trino/server/TaskResource.java

```diff
@@ -261,12 +283,18 @@ public void getTaskStatus(
     @Produces(MediaType.APPLICATION_JSON)
     public void acknowledgeAndGetNewDynamicFilterDomains(
             @PathParam("taskId") TaskId taskId,
+            @QueryParam("nodeId") String requestNodeId,
             @HeaderParam(TRINO_CURRENT_VERSION) Long currentDynamicFiltersVersion,
             @Context UriInfo uriInfo,
             @Suspended AsyncResponse asyncResponse)
     {
         requireNonNull(taskId, "taskId is null");
         requireNonNull(currentDynamicFiltersVersion, "currentDynamicFiltersVersion is null");
+        if (!nodeIdentifier.equals(requestNodeId)) {
+            asyncResponse.resume(Response.status(Status.BAD_REQUEST)
```

Comment:
return here

Reply:
That's why I don't like asyncResponse. You can only resume once but code doesn't enforce that :/